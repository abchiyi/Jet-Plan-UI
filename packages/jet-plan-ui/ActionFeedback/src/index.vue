<script lang="ts">
import baseAction from "./baseAction.vue";
import { h, defineComponent, Transition } from "vue";

const Mask = defineComponent({
  props: {
    styleConfig: {
      type: Object,
      require: true,
    },
  },
  render() {
    return h("span", {
      class: ["j-mask", "mask"],
      style: this.styleConfig,
    });
  },
});

export default defineComponent({
  name: "j-action-feedback",
  props: {
    name: { type: String, default: "j" }, //TODO 未添加
    hover: { type: Boolean, default: false },
    active: { type: Boolean, default: false },
    activeColor: {
      type: String,
      default: "var(--mask)",
    },
    activeOpacity: {
      type: Number,
      default: 1,
    },
  },
  data() {
    return {
      activated: false,
    };
  },
  methods: {
    handleAction(trigger: boolean) {
      if (!this.active) return;
      this.activated = trigger;
    },

    genMaskConfig() {
      return {
        background: this.activeColor,
        "--opacity": this.activeOpacity,
      };
    },
  },
  render() {
    const styleConfig = this.genMaskConfig();
    const Active = this.activated;
    const MaskActive = (() => {
      return h(
        Transition,
        { name: "mask" },
        {
          default() {
            if (!Active) return;
            return h(Mask, { styleConfig });
          },
        }
      );
    })();

    return h(
      baseAction,
      {
        class: [
          "j-action-feedback",
          this.hover ? "hover" : "",
          this.activated ? "active" : "",
        ],
        onActive_from: () => {
          this.handleAction(true);
        },
        onActive_to: () => {
          this.handleAction(false);
        },
      },
      {
        default: () => [this.$slots.default?.(), MaskActive],
      }
    );
  },
});
</script>
<style>
.j-action-feedback {
  outline: solid 3px #00000000;
  position: relative;
  transition: unset;
  overflow: hidden;
}

.j-action-feedback,
.j-action-feedback::after {
  transition: 0.3s var(--ease-out-slow);
}

.j-action-feedback::after {
  pointer-events: none;
  position: absolute;
  content: "";
  opacity: 0;
  bottom: 0;
  right: 0;
  left: 0;
  top: 0;
}

/* Mouse hover */
.j-action-feedback.hover:hover::after {
  background-color: var(--mask);
}

.j-action-feedback.hover::after {
  opacity: 0.5;
}

@supports (-webkit-tap-highlight-color: #ffffff00) {
  .j-action-feedback {
    -webkit-tap-highlight-color: #ffffff00;
  }
}

/* Mask */
.j-action-feedback .j-mask {
  transform: translate3d(0, 0, 0);
  opacity: var(--opacity);
  pointer-events: none;
  position: absolute;
  user-select: none;
  display: block;
  height: 100%;
  width: 100%;
  left: 0;
  top: 0;
}

.j-action-feedback .mask-leave-to {
  transition: 0.4s cubic-bezier(0.6, 0.6, 0.4, 0.7);
  opacity: 0;
}
</style>
