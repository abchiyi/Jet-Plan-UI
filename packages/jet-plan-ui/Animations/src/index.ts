import { h, Transition, defineComponent } from "vue";
import { validatorRange } from "../../tool";

export const TransitionFade = defineComponent({
  name: "j-transition-fade",
  render() {
    return h(
      Transition,
      { name: "j-transition-fade" },
      { default: () => [this.$slots.default?.()] }
    );
  },
});

const scaleName = "j-transition-scale";
export const TransitionScale = defineComponent({
  name: scaleName,
  render() {
    return h(
      Transition,
      { name: scaleName },
      { default: () => this.$slots.default?.() }
    );
  },
});

const foldedName = "j-transition-folded";
export const TransitionFolded = defineComponent({
  name: foldedName,
  methods: {
    enter(el: Element) {
      (el as HTMLElement).style.height = el.scrollHeight + "px";
    },
    afterEnter(el: Element) {
      (el as HTMLElement).style.height = "";
    },
    leave(el: Element) {
      const style = window.getComputedStyle(el);
      const paddingTop = Number(style.paddingTop.split("px")[0]);
      const paddingBottom = Number(style.paddingBottom.split("px")[0]);
      (el as HTMLElement).style.height =
        el.scrollHeight - paddingTop - paddingBottom + "px";
    },
    afterLeave(el: Element) {
      (el as HTMLElement).style.height = "";
    },
  },
  render() {
    return h(
      Transition,
      {
        name: foldedName,
        onEnter: this.enter,
        onAfterEnter: this.afterEnter,
        onLeave: this.leave,
        onAfterLeave: this.afterLeave,
      },
      { default: () => this.$slots.default?.() }
    );
  },
});

const sliderName = "j-transition-slide";
export const TransitionSlider = defineComponent({
  name: sliderName,
  props: {
    position: {
      type: String,
      required: true,
      validator: validatorRange(["left", "right", "top", "bottom"]),
    },
    opacity: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    style() {
      function position(str: string) {
        const position = {
          left: "-100%,0,0",
          right: "100%,0,0",
          top: "0,-100%,0",
          bottom: "0,100%,0",
        };
        switch (str) {
          case "left":
            return position.left;
          case "right":
            return position.right;
          case "top":
            return position.top;
          case "bottom":
            return position.bottom;
        }
      }
      return {
        "--migration-distance": `translate3d(${position(this.position)})`,
        "--opacity": this.opacity ? 0 : 1,
      };
    },
  },
  methods: {},
  render() {
    return h(
      Transition,
      {
        name: "j-transition-slide",
        style: this.style,
      },
      {
        default: () => this.$slots.default?.(),
      }
    );
  },
});
