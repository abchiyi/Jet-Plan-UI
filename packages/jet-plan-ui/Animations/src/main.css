/* Fade */
.j-transition-fade-leave-to,
.j-transition-fade-enter-from {
  opacity: 0;
}

.j-transition-fade-enter-active,
.j-transition-fade-leave-active {
  transition: opacity 0.4s var(--ease-out);
}

/* Folded */
.j-transition-folded-leave-to,
.j-transition-folded-enter-from {
  margin-bottom: 0 !important;
  margin-top: 0 !important;
  padding-bottom: 0 !important;
  padding-top: 0 !important;
  border-bottom: 0 !important;
  border-top: 0 !important;
  height: 0 !important;
}

.j-transition-folded-leave-active,
.j-transition-folded-enter-active {
  transition: all 0.4s var(--ease-out) !important;
  transform: translate3d(0, 0, 0);
  backface-visibility: hidden;
  overflow: hidden;
  display: block;
}

/* Scale */
.j-transition-scale-enter-active,
.j-transition-scale-leave-active {
  transition: all 0.4s var(--ease-out);
}

.j-transition-scale-enter-from,
.j-transition-scale-leave-active {
  transform: scale(0) !important;
  opacity: 0;
}

/* Slider */
.j-transition-slide-enter-from,
.j-transition-slide-leave-active {
  transform: var(--migration-distance);
  opacity: var(--opacity);
}

.j-transition-slide-enter-active,
.j-transition-slide-leave-active {
  transition: all 500ms var(--ease-in-out) !important;
}