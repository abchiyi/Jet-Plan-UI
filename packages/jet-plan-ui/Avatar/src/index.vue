<template>
  <div :class="classes">
    <div class="content">
      <slot></slot>
    </div>
  </div>
</template>

<script lang="ts">
import { JET_THEME } from "jet-plan-ui/theme";
import { Bumper, autoTextColor } from "jet-plan-ui/tool";
import { defineComponent } from "vue";
export default defineComponent({
  name: "j-avatar",
  props: {
    circle: {
      type: Boolean,
      default: true,
    },
  },
  setup() {
    return {
      colorBackground: JET_THEME.Theme.background.light,
    };
  },
  data() {
    return {
      colorText: "",
      Bumper: new Bumper(20),
    };
  },
  mounted() {
    this.Bumper.setAlarmCooled(() => {
      this.colorText = autoTextColor(this.$el);
    });
    this.Bumper.action();
  },
  updated() {
    this.Bumper.action();
  },
  computed: {
    classes() {
      return ["j-avatar", this.circle ? "circle" : ""];
    },
  },
});
</script>

<style>
.j-avatar {
  background: v-bind("colorBackground");
  color: v-bind(colorText);
  display: inline-block;
  position: relative;
  font-size: 1.5rem;
  user-select: none;
  overflow: hidden;
  min-height: 2em;
  min-width: 2em;
  margin: 5px;
}

.j-avatar .content {
  justify-content: center;
  align-items: center;
  position: absolute;
  display: flex;
  bottom: 0;
  right: 0;
  left: 0;
  top: 0;
}

.j-avatar.circle img,
.j-avatar.circle {
  border-radius: 2em;
}

.j-avatar img,
.j-avatar {
  border-radius: var(--m-radius);
}

.j-avatar img {
  height: var(--size);
  width: var(--size);
}
</style>
