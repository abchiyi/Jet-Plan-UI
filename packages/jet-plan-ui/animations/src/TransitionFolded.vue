<script>
import { h, Transition } from 'vue';
export default {
    name: 'j-transition-folded',
    methods: {
        enter(el) {
            el.style.height = el.scrollHeight + 'px';
        },
        afterEnter(el) {
            el.style.height = '';
        },

        leave(el) {
            let style = window.getComputedStyle(el);
            let paddingTop = style.paddingTop.split('px')[0];
            let paddingBottom = style.paddingBottom.split('px')[0];
            el.style.height =
                el.scrollHeight - paddingTop - paddingBottom + 'px';
        },
        afterLeave(el) {
            el.style.height = '';
        },
    },
    render() {
        return h(
            Transition,
            {
                name: 'j-transition-folded',
                onEnter: this.enter,
                onAfterEnter: this.afterEnter,

                onLeave: this.leave,
                onAfterLeave: this.afterLeave,
            },
            {
                default: () => this.$slots.default(),
            }
        );
    },
};
</script>

<style>
.j-transition-folded-leave-to,
.j-transition-folded-enter-from {
    margin-bottom: 0 !important;
    margin-top: 0 !important;
    padding-bottom: 0 !important;
    padding-top: 0 !important;
    height: 0 !important;
}
.j-transition-folded-leave-active,
.j-transition-folded-enter-active {
    transition: all 0.4s var(--ease-out) !important;
    transform: translate3d(0, 0, 0);
    backface-visibility: hidden;
    overflow: hidden;
    display: block;
}
</style>
