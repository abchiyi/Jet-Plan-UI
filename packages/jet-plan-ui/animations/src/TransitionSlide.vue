<script>
	import { h, Transition } from 'vue';
	export default {
		name: 'j-transition-slide',
		props: {
			position: {
				type: String,
				required: true,
				validator: function (value) {
					const checkout = ['left', 'right', 'top', 'bottom'];
					return checkout.indexOf(value.toLowerCase()) !== -1;
				},
			},
			opacity: {
				type: Boolean,
				default: true,
			},
		},
		computed: {
			style() {
				let position = {
					left: '-100%,0,0',
					right: '100%,0,0',
					top: '0,-100%,0',
					bottom: '0,100%,0',
				};
				return {
					'--migration-distance': `translate3d(${
						position[this.position]
					})`,
					'--opacity': this.opacity ? 0 : 1,
				};
			},
		},
		methods: {},
		render() {
			return h(
				Transition,
				{
					name: 'j-transition-slide',
					style: this.style,
				},
				{
					default: () => this.$slots.default(),
				}
			);
		},
	};
</script>

<style>
	.j-transition-slide-enter-from,
	.j-transition-slide-leave-active {
		transform: var(--migration-distance);
		opacity: var(--opacity);
	}
	.j-transition-slide-enter-active,
	.j-transition-slide-leave-active {
		transition: all 500ms var(--ease-in-out) !important;
	}
</style>
