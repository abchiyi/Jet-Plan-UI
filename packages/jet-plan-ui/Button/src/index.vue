<script lang="ts">
import { h, defineComponent, computed } from "vue";
import { ActionFeedback } from "../../ActionFeedback";
// TODO// import { autoTextColor } from "../../tool/lib";
import { JET_THEME } from "../../theme";
export default defineComponent({
  name: "j-button",
  setup() {
    const colors = computed(() => JET_THEME.Theme);

    return {
      colors,
    };
  },
  data() {
    return {
      value: false,
      maskSize: null,
      baseColor: "",
    };
  },
  mounted() {
    function checkAndSetTransition(el: HTMLElement) {
      function getTransitionDuration(el: HTMLElement) {
        return parseFloat(window.getComputedStyle(el).transitionDuration);
      }
      if (!getTransitionDuration(el)) {
        el.style.transition = "0.1s ease-in-out";
      }
    }
    const el = this.$el as HTMLElement;
    el.addEventListener("transitionend", () => {
      checkAndSetTransition(el);
      this.calcTextColor();
    });
    this.calcTextColor();
  },
  updated() {
    this.calcTextColor();
  },
  props: {
    disabled: {
      type: Boolean,
      default: false,
    },
    text: {
      type: Boolean,
      default: false,
    },
    primary: {
      type: Boolean,
      default: false,
    },
    block: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classes() {
      return [
        "shape",
        "j-button",
        this.text ? "text-button" : "button",
        this.disabled ? "disabled" : null,
        this.primary ? "primary" : null,
        this.block ? "block" : null,
      ];
    },
  },
  methods: {
    calcTextColor() {
      // TODO// import { autoTextColor } from "../../tool/lib";
      // if (!this.text) {
      //   this.baseColor = autoTextColor(this.$refs.self.$el);
      // }
    },
  },
  render() {
    return h(
      ActionFeedback,
      {
        class: this.classes,
        tag: "button",
        active: true,
        ref: "self",
        style: {
          "--base-text-color": this.baseColor,
        },
      },
      {
        default: () => [this.$slots.default?.()],
      }
    );
  },
});
</script>
<style>
/*------------ Default ----------*/

.j-button {
  background: v-bind("colors.background.dark");
  transition: all 0.3s var(--ease-out);
  color: v-bind("colors.text.default");
  box-sizing: border-box;
  display: inline-block;
  white-space: nowrap;
  position: relative;
  border-radius: 4px;
  font-size: 0.8rem;
  padding: 5px 10px;
  user-select: none;
  cursor: pointer;
}

.j-button {
  margin: calc(1em * 0.15);
}

/*------------ disabled ----------*/

.j-button.button.disabled {
  background: v-bind("colors.background.default");
  color: v-bind("colors.text.disabled");
  cursor: not-allowed;
}

/*------------ OneLine ----------*/
.j-button.block {
  margin-left: unset;
  margin-right: unset;
  display: block;
  width: 100%;
}

/* Button Mode Only*/
/*------------ Primary ----------*/
.j-button.primary {
  background-color: v-bind("colors.infoColors.primary.default");
  color: v-bind("colors.text.light");
}

/* Text Button */
/*------------ Default ----------*/
.j-button.text-button {
  background: unset;
  padding: 6px 8px;
  outline: unset;
}
/*------------ Primary ----------*/
.j-button.text-button.primary {
  color: v-bind("colors.infoColors.primary.default");
}

/* ------------ disabled ---------- */
.j-button.text-button.disabled {
  color: v-bind("colors.text.disabled");
  background: unset;
}

/* p span */
span > .j-button.text-button,
p > .j-button.text-button {
  padding-top: unset;
  padding-bottom: unset;
}
</style>
