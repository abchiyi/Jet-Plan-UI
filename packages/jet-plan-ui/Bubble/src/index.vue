<script lang="ts">
import baseAction from "../../ActionFeedback/src/baseAction.vue";
import { TransitionSlider } from "../../Animations";
import { Row } from "../../Grid";
import { defineComponent, h, type PropType } from "vue";
import { Bumper, getOffset, getEl } from "../../tool";
export default defineComponent({
  name: "j-bubble",
  props: {
    position: {
      type: String as PropType<
        | "top"
        | "top-end"
        | "top-start"
        | "bottom"
        | "bottom-end"
        | "bottom-start"
        | "left"
        | "left-top"
        | "left-bottom"
        | "right"
        | "right-top"
        | "right-bottom"
      >,
      default: "top",
    },

    show: {
      type: Boolean,
      default: undefined,
    },
  },

  data() {
    return {
      // 初始值同步自 prop show 以确保气泡正常显示
      showBubble: this.show,
      BumperDisplay: new Bumper(350),
      BumperHide: new Bumper(50),
      newPosition: this.position,
    };
  },
  mounted() {
    // 设置气泡展示&消失演示避免闪烁
    this.BumperDisplay.setAlarmCooled(() => (this.showBubble = true));
    this.BumperHide.setAlarmCooled(() => (this.showBubble = false));

    this.setEvent();
  },
  computed: {
    ClassBubble() {
      return ["bubble", ...this.newPosition.split("-")];
    },
    positionReverse() {
      const [position] = this.newPosition.split("-");
      switch (position) {
        case "top":
          return "bottom";
        case "bottom":
          return "top";
        case "left":
          return "right";
        case "right":
          return "left";
        default: // 默认块解决类型分析问题，实际运行将不会被执行
          return "top";
      }
    },
  },
  methods: {
    displayBubble() {
      this.edgeAvoidance();
      this.BumperDisplay.action();
    },
    hideBubble() {
      this.BumperHide.action();
    },
    edgeAvoidance() {
      // 计算元素的每条边距离屏幕边缘是否小于预设值
      const nearEdgeOfScreen = (() => {
        const marginMax = 50;
        let elSize = getOffset(this.$el);
        return {
          top: elSize.size.top <= marginMax,
          left: elSize.size.left <= marginMax,
          right: window.innerWidth - elSize.size.right <= marginMax,
          bottom: window.innerHeight - elSize.size.bottom <= marginMax,
        };
      })();

      let [position, start] = this.position.split("-");

      function findNewPosition() {
        switch (true) {
          case "top" != position && !nearEdgeOfScreen.top:
            return "top";
          case "bottom" != position && !nearEdgeOfScreen.bottom:
            return "bottom";
          case "left" != position && !nearEdgeOfScreen.left:
            return "left";
          case "right" != position && !nearEdgeOfScreen.right:
            return "right";
        }
        return "canter";
      }

      function NewStart() {
        switch (true) {
          case nearEdgeOfScreen.left:
            return "start";
          case nearEdgeOfScreen.right:
            return "end";
          case nearEdgeOfScreen.left && nearEdgeOfScreen.right:
            return "";
        }
        return start;
      }

      function NewPosition() {
        switch (true) {
          case "top" == position && nearEdgeOfScreen.top:
            if (!nearEdgeOfScreen.bottom) {
              return "bottom";
            } else {
              return findNewPosition();
            }
          case "bottom" == position && nearEdgeOfScreen.bottom:
            if (!nearEdgeOfScreen.top) {
              return "top";
            } else {
              return findNewPosition();
            }
          case "left" == position && nearEdgeOfScreen.left:
            if (!nearEdgeOfScreen.right) {
              return "right";
            } else {
              return findNewPosition();
            }
          case "right" == position && nearEdgeOfScreen.right:
            if (!nearEdgeOfScreen.left) {
              return "left";
            } else {
              return findNewPosition();
            }
          default:
            return position;
        }
      }

      this.newPosition = (
        start ? `${NewPosition()}-${NewStart()}` : NewPosition()
      ) as any;
    },
    /**
     * 在 this.show 不为 undefined 时不设置事件触发气泡显示
     */
    setEvent() {
      const el = getEl(this, "j-bubble");
      if (this.show == undefined) {
        el.addEventListener("mouseenter", this.displayBubble);
        el.addEventListener("mouseleave", this.hideBubble);
      } else {
        el.removeEventListener("mouseenter", this.displayBubble);
        el.removeEventListener("mouseleave", this.hideBubble);
      }
    },
  },
  watch: {
    position(v) {
      this.newPosition = v;
    },
    show(v) {
      // 使用 Prop 'show' 控制气泡是否显示
      this.setEvent();
      v ? this.displayBubble() : this.hideBubble();
    },
  },
  render() {
    const Bubble = h(
      "div",
      {
        class: this.ClassBubble,
      },
      this.$slots.bubble?.()
    );
    return h(
      baseAction,
      {
        class: "j-bubble",
      },
      {
        default: () =>
          h(
            Row,
            { X: "center", Y: "center" },
            {
              default: () => [
                this.$slots.default?.(),
                h(
                  TransitionSlider,
                  { position: this.positionReverse },
                  {
                    default: () => (this.showBubble ? Bubble : undefined),
                  }
                ),
              ],
            }
          ),
      }
    );
  },
});
</script>

<style scoped>
.j-bubble {
  display: inline-block;
  position: relative;
}

.j-bubble .bubble {
  background: rgb(66, 66, 66);
  border-radius: var(--m-radius);
  /* color: var(--base-text-color); */
  box-sizing: border-box;
  display: inline-block;
  padding: 5px 10px;
  user-select: none;
  font-size: 0.8rem;
  white-space: nowrap;
  position: absolute;
  z-index: 1;
  pointer-events: none;
}

.j-bubble .bottom.start,
.j-bubble .top.start {
  left: 0;
}

.j-bubble .bottom.end,
.j-bubble .top.end {
  right: 0;
}

.j-bubble .top {
  bottom: 100%;
  margin-bottom: 5px;
}

.j-bubble .bottom {
  top: 100%;
  margin-top: 5px;
}

.j-bubble .right.top,
.j-bubble .left.top {
  top: 0;
  bottom: unset;
}

.j-bubble .right.bottom,
.j-bubble .left.bottom {
  bottom: 0;
  top: unset;
}

.j-bubble .right {
  left: 100%;
  margin-left: 5px;
}

.j-bubble .left {
  right: 100%;
  margin-right: 5px;
}

.j-bubble .right.j-transition-slide-enter-from,
.j-bubble .right.j-transition-slide-leave-active {
  transform: translate3d(-40px, 0, 0);
}

.j-bubble .left.j-transition-slide-enter-from,
.j-bubble .left.j-transition-slide-leave-active {
  transform: translate3d(40px, 0, 0);
}
</style>
