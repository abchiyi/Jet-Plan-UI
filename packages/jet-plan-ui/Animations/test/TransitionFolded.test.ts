import { mount } from "@vue/test-utils";
import { expect, describe, test } from "vitest";
import { TransitionFolded } from "../src";

describe("TransitionFoldedTest", () => {
  test("Slot:default", () => {
    const slotContent = "Default slot";
    const wrapper = mount(TransitionFolded, {
      slots: {
        default: () => {
          return slotContent;
        },
      },
    });
    expect(wrapper.html()).matchSnapshot();
    expect(wrapper.text()).toBe(slotContent);
  });
  test("Function:enter", () => {
    const wrapper = mount(TransitionFolded);
    const El = document.createElement("div");
    wrapper.vm.enter(El);

    expect(El.scrollHeight).toEqual(Number(El.style.height.split("px")[0]));
  });

  test("Function:afterEnter", () => {
    const wrapper = mount(TransitionFolded);
    const El = document.createElement("div");
    wrapper.vm.afterEnter(El);

    expect(El.style.height).toBe("");
  });

  test("Function:leave", () => {
    const wrapper = mount(TransitionFolded);
    const El = document.createElement("div");
    wrapper.vm.leave(El);

    expect(El.style.height).toBe("0px");
  });
  test("Function:afterLeave", () => {
    const wrapper = mount(TransitionFolded);
    const El = document.createElement("div");
    wrapper.vm.afterLeave(El);

    expect(El.style.height).toBe("");
  });
});
