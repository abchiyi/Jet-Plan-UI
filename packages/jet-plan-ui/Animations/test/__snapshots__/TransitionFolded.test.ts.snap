// Vitest Snapshot v1

exports[`TransitionFoldedTest > Slot:default 1`] = `"<transition-stub name=\\"j-transition-folded\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">Default slot</transition-stub>"`;
